frontend/catalyst/pennylane_extensions.py

```diff
@@ -2113,3 +2116,286 @@ def _callee():
+) -> Callable:
+    """A :func:`~.qjit` compatible vectorizing map.
+    Creates a function which maps an input function over argument axes.
+
+    Args:
+        f (Callable): A Python function containing PennyLane quantum operations.
+        in_axes (Union[int, Sequence[Any]]): Specifies the value(s) over which input
+            array axes to map.
+        out_axes (Union[int, Sequence[Any]]): Specifies where the mapped axis should appear
+            in the output.
+        axis_size (int): An integer can be optionally provided to indicate the size of the
+            axis to be mapped. If omitted, the size of the mapped axis will be inferred from
+            the provided arguments.
+
+    Returns:
+        Callable: Vectorized version of ``fn``.
+
+    Raises:
+        ValueError: Invalid ``in_axes``, ``out_axes``, and ``axis_size`` values.
+
+
+    **Example**
+
+    .. code-block:: python
+
+        @qjit
+        def workflow(x, y, z):
+            @qml.qnode(qml.device(backend, wires=1))
+            def circuit(x, y):
+                qml.RX(jnp.pi * x[0] + y, wires=0)
+                qml.RY(x[1] ** 2, wires=0)
+                qml.RX(x[1] * x[2], wires=0)
+                return qml.expval(qml.PauliZ(0))
+
+            def postcircuit(y, x, z):
+                return circuit(x, y) * z
+
+            res = vmap(postcircuit, in_axes=(0, 0, None))(y, x, z)
+            return res
+
+        y = jnp.array([jnp.pi, jnp.pi / 2, jnp.pi / 4])
+        x = jnp.array(
+            [
+                [0.1, 0.2, 0.3],
+                [0.4, 0.5, 0.6],
+                [0.7, 0.8, 0.9],
+            ]
+        )
+
+    >>> workflow(x, y, 1)
+    [-0.93005586, -0.97165424, -0.6987465]
+
+    .. details::
+        :title: Selecting batching axes for arguments
+
+        The ``in_axes`` parameter provides different modes the allow large- and fine-grained
+        control over which arguments to apply the batching transformation on. Enabling batching for
+        a particular argument requires that the selected axis be of the same size as the determined
+        batch size, which is the same for all arguments.
+
+        The following modes are supported:
+
+        - ``int``: Specifies the same batch axis for all arguments
+        - ``Tuple[int]``: Specify a different batch axis for each argument
+        - ``Tuple[int | None]``: Same as previous, but selectively disable batching for certain
+          arguments with a ``None`` value
+        - ``Tuple[int | PyTree[int] | None]``: Same as previous, but specify a different batch
+          axis for each leaf of an argument (Note that the ``PyTreeDefs``, i.e. the container
+          structure, must match between the ``in_axes`` element and the corresponding argument.)
+        - ``Tuple[int | PyTree[int | None] | None]``: Same as previous, but selectively disable
+          batching for individual PyTree leaves
+
+    .. details::
+        :title: Selecting output axes for return results
```

Comment:
I don't think this section adds a lot of value since it just mirrors the one above. Maybe we can just add a sentence that the same applies to outputs in the above section?

Reply:
@dime10 what about this update? [6a8a215](https://github.com/PennyLaneAI/catalyst/pull/497/commits/6a8a215b1d8e75122e5ee3e44b6f5d2f71a08c56)